atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterTests.java

```diff
@@ -0,0 +1,467 @@
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.dbkvs.impl.oracle;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import com.google.common.collect.Sets;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.dbkvs.ImmutableOracleDdlConfig;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleDdlConfig;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleTableNameGetter;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleTableNameGetterImpl;
+import com.palantir.atlasdb.keyvalue.dbkvs.cleaner.ImmutableOracleNamespaceDeleterParameters;
+import com.palantir.atlasdb.keyvalue.dbkvs.cleaner.OracleNamespaceDeleter;
+import com.palantir.atlasdb.keyvalue.dbkvs.cleaner.OracleNamespaceDeleter.OracleNamespaceDeleterParameters;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.ConnectionSupplier;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.DbKvs;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.OverflowMigrationState;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.TableValueStyleCache;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.oracle.SqliteOracleAdapter.TableDetails;
+import com.palantir.atlasdb.namespacedeleter.NamespaceDeleter;
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class OracleNamespaceDeleterTests {
+    @ClassRule
+    public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+
+    private static final String TABLE_PREFIX = "a_";
+    private static final String OVERFLOW_TABLE_PREFIX = "ao_";
+    private static final String ANOTHER_TABLE_PREFIX = "test_";
+    private static final String ANOTHER_OVERFLOW_TABLE_PREFIX = "over_";
+    private static final String TABLE_NAME_1 = "abc";
+    private static final String TABLE_NAME_2 = "hello";
+    private static final String TABLE_NAME_3 = "world";
+    private static final String TEST_USER = "testuser";
+    private static final String TEST_USER_2 = "testuser2";
+
+    private ExecutorService executorService;
+    private SqliteOracleAdapter sqliteOracleAdapter;
+
+    @Before
+    public void before() throws IOException {
+        executorService = MoreExecutors.newDirectExecutorService();
```

Comment:
Is there any reason we need this to be shared across invocations in the same test method? If not, it seems that you can just get away with passing this in wherever it's used